backends/smt2/smt2.cc

```diff
@@ -532,10 +532,10 @@ struct Smt2Worker
 			if (cell->type.in("$anyconst", "$anyseq", "$allconst", "$allseq"))
 			{
 				registers.insert(cell);
-				string infostr = cell->attributes.count("\\src") ? cell->attributes.at("\\src").decode_string().c_str() : get_id(cell);
+				string infostr;
 				if (cell->attributes.count("\\reg"))
-					infostr += " " + cell->attributes.at("\\reg").decode_string();
-				decls.push_back(stringf("; yosys-smt2-%s %s#%d %d %s\n", cell->type.c_str() + 1, get_id(module), idcounter, GetSize(cell->getPort("\\Y")), infostr.c_str()));
+					infostr += cell->attributes.at("\\reg").decode_string();
+				decls.push_back(stringf("; yosys-smt2-%s %s#%d %d %s %s\n", cell->type.c_str() + 1, get_id(module), idcounter, GetSize(cell->getPort("\\Y")), get_id(cell->getPort("\\Y").as_wire()), infostr.c_str()));
```

Comment:
There's value to providing a src attribute in the smt comment when present (this change completely removes that information) and there's no guarantee that `cell->getPort("\\Y").as_wire()` doesn't return a NULL pointer afaict.

Reply:
Thanks for the feedback.
Would you advice to add this info, if available, as an addition? Not sure anymore how this would affect retrieving of the info, I think I tried to align this with other cases.
Or does this mean changes to this comment should not be made?

Reply:
I think I'm not too picky about the details, just think this information should be present, and that `.as_wire()` is no good here for the reasons stated above.